euphoria-flink/src/main/java/cz/seznam/euphoria/flink/batch/BatchFlowTranslator.java

```diff
@@ -140,19 +143,27 @@ protected FlowOptimizer createOptimizer() {
     // translate each operator to proper Flink transformation
     dag.traverse().map(Node::get).forEach(op -> {
       Operator<?, ?> originalOp = op.getOriginalOperator();
-      Translation<Operator<?, ?>> tx = translations.get(originalOp.getClass());
-      if (tx == null) {
+      List<Translation> txs = this.translations.get(originalOp.getClass());
+      if (txs.isEmpty()) {
         throw new UnsupportedOperationException(
                 "Operator " + op.getClass().getSimpleName() + " not supported");
       }
       // ~ verify the flowToDag translation
-      Preconditions.checkState(
-          tx.accept == null || Boolean.TRUE.equals(tx.accept.apply(originalOp)));
-
-      DataSet<?> out = tx.translator.translate(op, executorContext);
-
-      // save output of current operator to context
-      executorContext.setOutput(op, out);
+      Translation<Operator<?, ?>> firstMatch = null;
+      for (Translation tx : txs) {
+        if (tx.accept == null || Boolean.TRUE.equals(tx.accept.apply(originalOp))) {
+          firstMatch = tx;
+          break;
```

Comment:
We should document and emphasize this behavior, that first match will be the one that will be called during the translation.

Reply:
I have a feeling that this translation "interface" is very similar between runtimes, so it would be nice to abstract it into euphoria-core, but it should be done in a separate PR / issue